src/Shared/FileUtilities.cs

```diff
@@ -532,14 +532,23 @@ internal static string GetDirectory(string fileSpec)
         /// <returns></returns>
         internal static bool HasExtension(string fileName, string[] allowedExtensions)
         {
-            string fileExtension = Path.GetExtension(fileName);
-            foreach (string extension in allowedExtensions)
+            Debug.Assert(allowedExtensions != null && allowedExtensions.Length > 0);
+
+            // Easiest way to invoke invalid path chars
+            // check, which callers are relying on.
+            if (Path.HasExtension(fileName))
             {
-                if (String.Compare(fileExtension, extension, StringComparison.CurrentCultureIgnoreCase) == 0)
+                foreach (string extension in allowedExtensions)
                 {
-                    return true;
+                    Debug.Assert(!String.IsNullOrEmpty(extension) && extension[0] == '.');
```

Comment:
IIUC, this (method) is meant for internal use but gets exposed via RAR's extension checking (or others in future?). Isn't this causing a slight difference in behavior now, because earlier it would accept `extension` to be only `.foo` but now it would even accept `.bar.foo`? Just thinking aloud. If it is fine or we don't care about it then maybe we should add a test for this case too?

Reply:
And similar concern about the `'.'` check too - is this something we care about at runtime? Or maybe we want to be lax on that because it is rare to not have the `'.'` and the performance concern is more important.

Reply:
@cdmihai Did you reply to these concerns before merging? These look like legitimate concerns.

Reply:
Initially I thought it would be OK, but now I realize it will catch files like `foo.config.exe`, which is not OK. I'll fix this up to only compare starting from the last '.' and send a PR.
Regarding the `.`, the original code was not checking for it, so I don't whether it would be good to start failing it now.

Reply:
Ah, I remember now. With both the previous version of the code and the current version, `.exe` would still catch `foo.appconfig.exe`, because `Path.GetExtension("foo.appconfig.exe")` returns ".exe". The only difference now is that before, if you told RAR to start collecting extensions like ".config.exe" it would not get them, but now it would. So this is only a change in behaviour if somebody modified the inputs to RAR to include multi dot extensions and expected RAR to not find them, which means they expect RAR to do a no-op for those inputs. Since that seems plausible, and since it does not affect RAR's output, I decided it's fine.

Let me know if I missed something.

Reply:
> before, if you told RAR to start collecting extensions like ".config.exe" it would not get them, but now it would.

Yep, that is exactly what I was talking about.

> So this is only a change in behaviour if somebody modified the inputs to RAR to include multi dot extensions ...

Makes sense. Probably nitpicking: But if it is so specific to RAR, then maybe it shouldn't be in `FileUtilities`. Or should be renamed
to reflect that. Or add a comment to warn future non-RAR callers.

Reply:
Or replace the other potentially existing extension checking code clones to reuse this one :)